cli/lib/nf3.mjs

```diff
@@ -194,9 +192,10 @@ class Nf3 {
     // we need a Mutex so that we don't get a nonce-updating race.
 
     let tx;
+    let nonce;
     await this.nonceMutex.runExclusive(async () => {
-      this.nonce = await this.web3.eth.getTransactionCount(this.ethereumAddress);
-
+      nonce = await this.web3.eth.getTransactionCount(this.ethereumAddress);
+      nonce += this.txsNotConfirmed; // we need to add the number of transactions not confirmed yet
```

Comment:
In a previous PR it was introduced the change of `this.web3.eth.getTransactionCount(this.ethereumAddress)` for `this.nonce`every time we submit a transaction. So `this.nonce` then has no sense as it always is being set to `this.web3.eth.getTransactionCount(this.ethereumAddress)`. That's why I removed this from the class and put it as local variable. If you look at master branch you will see it. I think this was introduced because errors when revert from contract, etc. This `this.nonce` was not decremented and produced nonce errors because nonce account was different from `this.nonce` that was previously incremented.